src/core/lib/event_engine/iomgr_engine/ev_epoll1_linux.cc

```diff
@@ -487,24 +506,26 @@ void Epoll1EventHandle::SetWritable() { write_closure_->SetReady(); }
 
 void Epoll1EventHandle::SetHasError() { error_closure_->SetReady(); }
 
-absl::Status Epoll1Poller::Work(grpc_core::Timestamp deadline,
-                                std::vector<EventHandle*>& pending_events) {
+Poller::WorkResult Epoll1Poller::Work(EventEngine::Duration timeout) {
+  Poller::Events pending_events;
+  grpc_core::Timestamp deadline = ToTimestamp(timeout);
   if (g_epoll_set_.cursor == g_epoll_set_.num_events) {
-    auto status = DoEpollWait(deadline);
-    if (!status.ok()) {
-      return status;
+    if (DoEpollWait(deadline) == 0) {
```

Comment:
This converts a timeout to a deadline, then `DoEpollWait` converts the deadline back to a timeout. I suggest changing the signature to `DoEpollWait(EventEngine::Duration timeout)`.

Please review and use this: https://github.com/grpc/grpc/pull/30427. I forked it off from my IOCP poller branch, I think it will work nicely here.

Reply:
Done.